src/me/corriekay/pokegoutil/windows/PokemonTab.java

```diff
@@ -198,13 +198,17 @@ private void renameSelected() {
             if (result.getNumber() == NicknamePokemonResponse.Result.SUCCESS_VALUE) {
                 success.increment();
                 System.out.println("Renaming " + PokeHandler.getLocalPokeName(pokemon) + " from \"" + pokemon.getNickname() + "\" to \"" + PokeHandler.generatePokemonNickname(renamePattern, pokemon) + "\", Result: Success!");
+            } else if (result.getNumber() == NicknamePokemonResponse.Result.UNSET_VALUE) { // FIXME: See PokeHandler.java@97
+                success.increment();
+                System.out.println("Not renaming " + PokeHandler.getLocalPokeName(pokemon) + ", already named " + PokeHandler.generatePokemonNickname(renamePattern, pokemon));
             } else {
                 err.increment();
                 System.out.println("Renaming " + PokeHandler.getLocalPokeName(pokemon) + " failed! Code: " + result.toString() + "; Nick: " + PokeHandler.generatePokemonNickname(renamePattern, pokemon));
             }
 
-            // If not last element, sleep until the next one
-            if (!selection.get(selection.size() - 1).equals(pokemon)) {
+            // If not last element, sleep until the next one and the rename was successful
+            if (!selection.get(selection.size() - 1).equals(pokemon)
+                    && result.getNumber() == NicknamePokemonResponse.Result.SUCCESS_VALUE) {
```

Comment:
Can you change it to `result.getNumber() == NicknamePokemonResponse.Result.UNSET_VALUE`?
This delay is made for not querying the server too frequently, so it should wait if the Server query failed too.
Just skip the delay when the nickname was skipped.